src/vaadin-custom-field-mixin.html

```diff
@@ -144,9 +141,24 @@
       this.__settingValue = false;
     }
 
-    __setInputsFromChildNodes() {
-      const inputs = Array.from(this.querySelectorAll('*')).filter(node => node.validate || node.checkValidity);
-      this._setInputs(inputs);
+    __getInputsFromSlot() {
+      const isInput = (node => node.validate || node.checkValidity);
+      const slottedElements = Polymer.FlattenedNodesObserver.getFlattenedNodes(this.$.slot)
```

Comment:
Note: you can replace ` Polymer.FlattenedNodesObserver.getFlattenedNodes(this.$.slot)` with the following: `this.$.slot.assignedNodes({flatten: true});` as this is what Polymer does under the hood.